### DISCUSSION TITLE
zfind using participle

### DISCUSSION THREAD

**Discussion by laktak** [alecthomas/participle#410] [Show and tell]
Hi! I had a lot of fun building [zfind](https://github.com/laktak/zfind) using participle :)

I took the SQL example and modified it into a 'where' expression parser and evaluator. If this can help anyone as an example the code is here: https://github.com/laktak/zfind/tree/master/filter.

I made some changes to the syntax to make `not` work for [`between` and `in`](https://github.com/laktak/zfind/blob/825bf113579d9f1508a85cf285feb3475cf31524/filter/lang.go#L31) as well as [like](https://github.com/alecthomas/participle/blob/5f96b0729ffe780f158a850dadf0b523610e93be/_examples/sql/main.go#L73) from the original.

**Comment by alecthomas** [comment]
Awesome, zfind looks cool :)
